src/components/CheckAuth/CheckAuth.js

```diff
@@ -117,7 +167,10 @@ class CheckPin extends React.Component<Props, State> {
     const options = {
       mnemonic: revealMnemonic,
     };
-    checkPin(pin, onPinValid, options);
+    checkPin(pin, (_, wallet) => {
+      onPinValid(_, wallet);
+      this.setState({ showPin: false });
+    }, options);
```

Comment:
Just thinking, if we want to call `onPinValid` and hide PIN input then it would be good to make sure state is updated and then callback is called.
const onPinValidSuccess = (...params) => this.setState({ showPin: false }, () => { onPinValid(...params); })
```suggestion
    checkPin(pin, onPinValidSuccess, options);
```